### DISCUSSION TITLE
GalaxyBudsClient for IOS or MacOS

### DISCUSSION THREAD

**Discussion by PedroLuisBrilhadori** [timschneeb/GalaxyBudsClient#414] [General]
Is it possible to adapt or make this software for IOS or macOS devices? I have an Iphone X and Samsung's "Samsung Galaxy Buds" app doesn't work for the Galaxy buds pro

I'd like to contribute to this. But, I need learn more, can someone help me? It's possible?

**Comment by timschneeb** [comment]
An iOS version is generally impossible (without a jailbreak) because the Buds mainly use Bluetooth BR/EDR, which Apple heavily restricts. It would require Samsung to license and embed [Apple's proprietary DRM MFi chip](https://stackoverflow.com/a/28473622/11559090) into the earbuds.

Only Bluetooth LE is available without restrictions; however, only the older earbud models have this additional compatibility with iOS, but they dropped it for more important features starting with the Buds Pro.

There was a macOS port in progress; however, my old 2011 Macbook completely died at some point, and I didn't recover the data from its hard drive yet. Because of that, the macOS port is abandoned (from my side). The work on the UI (C#) was mostly done, but the native Bluetooth backend implementation (written in Objective-C++) is unfinished. Also, only Intel Macs were targeted. If you want to continue it, I can try to retrieve the data, but I'm not quite sure if my current machine even has a fitting connector for the HDD :sweat_smile:

**Reply by PedroLuisBrilhadori** [reply]
For now I don't have a macos available. But anyway, thanks for the reply!

**Reply by mediaexplorer74** [reply]
Hello Pedro,

Maybe, you have some old sweet "5-cents-cost" retrospective phone Lumia? ;))) UWP app are welcome :)
